src/libs/feature-previews-config.js

```diff
@@ -0,0 +1,16 @@
+const featurePreviewsConfig = [
+  {
+    id: 'data-table-versioning',
+    title: 'Data Table Versioning',
+    description: 'Enabling this feature will allow you to save uniquely named versions of data tables. These saved versions will appear in the Data tab and can be restored at any time.',
+    groups: ['preview-data-versioning-and-provenance']
+  },
+  {
+    id: 'data-table-provenance',
+    title: 'Data Table Provenance',
+    description: 'Enabling this feature will allow you to view information about the workflow that generated data table columns and files.',
+    groups: ['preview-data-versioning-and-provenance']
+  }
+]
```

Comment:
This is unnecessarily verbose since it has the ID effectively duplicated. If you remove that duplication, you're left with something like:

```
isFeaturePreviewEnabled(featurePreviewConfig.myFeature)
```

Which is equally prone to misspelling.

I'm looking forward to a Typescript future where we don't have to worry about stuff like this.

Reply:
I don't think the potential issues I was trying to highlight would be solved by static type-safety (which I like myself FWIW). For example, on this branch, there are three occurrences of `isFeaturePreviewEnabled('data-table-provenance')`. This means, if we rename the feature to, say, `data-table-sources`, all those places need to be changed, and they need to be changed carefully, because if I misspell one as `data-table-source`, it'd pass type checking but not map to anything.

Reply:
With Typescript, this could be handled with either an [enum](https://www.typescriptlang.org/docs/handbook/enums.html#string-enums) or a [literal type](https://www.typescriptlang.org/docs/handbook/literal-types.html#string-literal-types).

```js
type FeatureId = 'feature-one' | 'feature-two'

const isFeaturePreviewEnabled = (id: FeatureId): boolean => ...
```